pkg/hs/urbit-king/lib/Urbit/Vere/Eyre/Serv.hs

```diff
@@ -254,19 +255,28 @@ startServer
   -> Net.Socket
   -> Maybe W.Port
   -> TVar E.LiveReqs
+  -> IO ()
   -> RIO e ()
-startServer typ hos por sok red vLive = do
+startServer typ hos por sok red vLive onFatal = do
   envir <- ask
 
   let host = case hos of
         SHLocalhost -> "127.0.0.1"
         SHAnyHostOk -> "*"
 
+  let handler r e
+        | Just (IOError {ioe_type = ResourceExhausted}) <- fromException e
+        = runRIO envir $ do
+            logError (displayShow e)
```

Comment:
Replying to your top-level comment: We do print the error before shutting down, and do so right here. Do you want this to be to stderr in addition to (or instead of) the logError?